src/test/java/com/dev9/presentation/service/ParallelStreamTest.java

```diff
@@ -0,0 +1,56 @@
+package com.dev9.presentation.service;
+
+
+import static org.fest.assertions.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ForkJoinPool;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.junit.Test;
+
+
+public class ParallelStreamTest {
+
+    @Test
+    public void test_in_line() {
+
+        final List<Integer> primes = IntStream.range(2, 1_000_000)
+                .boxed()
+                .parallel()
+                .filter(Prime::isPrime)
+                .collect(Collectors.toList());
+
+        assertThat(primes.size()).isEqualTo(78498);
+    }
+
+    @Test
+    public void test_parallel_supplied_forkJoinPool() throws ExecutionException, InterruptedException {
+
+        ForkJoinPool forkJoinPool = new ForkJoinPool(2);
+        List<Integer> primes = forkJoinPool.submit(
+                () ->
+                        IntStream.range(2, 1_000_000)
+                                .parallel()
+                                .boxed()
+                                .filter(Prime::isPrime)
+                                .collect(Collectors.toList())
+                ).get();
+
+        assertThat(primes.size()).isEqualTo(78498);
+    }
+
+
+    public static class Prime {
```

Comment:
Cribbed from Wikipedia (Wilson's theorem):

(p-1)! &#8801;-1(mod p)

Java implementation:

public static boolean isPrime(long n) {
    if (n <= 3) {
        return n > 1;
    } else if (n % 2 == 0 || n % 3 == 0) {
        return false;
    } else {
        for (int i = 5; i * i <= n; i += 6) {
            if (n % i == 0 || n % (i + 2) == 0) {
                return false;
            }
        }
        return true;
    }
}